src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php

```diff
@@ -0,0 +1,89 @@
+
+namespace Core\Infrastructure\Security\ProviderConfiguration\OpenId\Repository;
+
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Core\Application\Security\ProviderConfiguration\OpenId\Repository\WriteOpenIdConfigurationRepositoryInterface
+    as WriteRepositoryInterface;
+use Core\Domain\Security\ProviderConfiguration\OpenId\Model\OpenIdConfiguration;
+
+class DbWriteOpenIdConfigurationRepository extends AbstractRepositoryDRB implements WriteRepositoryInterface
+{
+    /**
+     * @param DatabaseConnection $db
+     */
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @param OpenIdConfiguration $configuration
+     */
+    public function updateConfiguration(OpenIdConfiguration $configuration): void
+    {
+        $statement = $this->db->prepare(
+            $this->translateDbName(
+                "UPDATE `:db`.`provider_configuration` SET
+                `custom_configuration` = :customConfiguration, `is_active` = :isActive, `is_forced` = :isForced
+                WHERE `name`='openid'"
+            )
+        );
+        $statement->bindValue(
+            ':customConfiguration',
+            json_encode($this->buildCustomConfigurationFromOpenIdConfiguration($configuration)),
+            \PDO::PARAM_STR
+        );
+        $statement->bindValue(':isActive', $configuration->isActive() ? '1' : '0', \PDO::PARAM_STR);
+        $statement->bindValue(':isForced', $configuration->isForced() ? '1' : '0', \PDO::PARAM_STR);
+        $statement->execute();
+    }
+
+    /**
+     * Format OpenIdConfiguration for custom_configuration.
+     *
+     * @param OpenIdConfiguration $configuration
+     * @return array<string, mixed>
+     */
+    private function buildCustomConfigurationFromOpenIdConfiguration(OpenIdConfiguration $configuration): array
+    {
+        return [
+            'trusted_client_addresses' => $configuration->getTrustedClientAddresses(),
+            'blacklist_client_addresses' => $configuration->getBlacklistClientAddresses(),
+            'base_url' => $configuration->getBaseUrl(),
+            'authorization_endpoint' => $configuration->getAuthorizationEndpoint(),
+            'token_endpoint' => $configuration->getTokenEndpoint(),
+            'introspection_token_endpoint' => $configuration->getIntrospectionTokenEndpoint(),
+            'userinfo_endpoint' => $configuration->getUserInformationsEndpoint(),
+            'endession_endpoint' => $configuration->getEndSessionEndpoint(),
+            'connection_scopes' => $configuration->getConnectionScopes(),
+            'login_claim' => $configuration->getLoginClaim(),
+            'client_id' => $configuration->getClientId(),
+            'client_secret' => $configuration->getClientSecret(),
+            'authentication_type' => $configuration->getAuthenticationType(),
+            'verify_peer' => $configuration->isVerifyPeer()
```

Comment:
should be updated in model too
```suggestion
            'verify_peer' => $configuration->verifyPeer()
```